runtime/onert/core/src/odc/CodegenManager.cc

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "CodegenLoader.h"
+#include "odc/CodegenManager.h"
+#include "util/Utils.h"
+
+#include <fstream>
+#include <mutex>
+
+namespace onert
+{
+namespace odc
+{
+
+bool CodegenManager::codegen(const char *target, CodegenPreference pref)
+{
+  // codegen function is thread-unsafe
+  static std::mutex lock;
+  std::lock_guard<std::mutex> guard(lock);
+
+  if (_export_model_path.empty())
+    throw std::runtime_error("Export model path is not set");
+
+  std::ifstream file(_model_path);
```

Comment:
this `file` seems to only check whether `model` exists or not. the variable `file` can be hidden by using `{ }`.

```cpp
{
  std::ifstream file(_model_path);
  ...
}
```